### PR TITLE
privilege,executor: update DBIsVisible() function for RBAC

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

```
	CREATE DATABASE app_db
	CREATE ROLE 'app_developer'
	GRANT ALL ON app_db.* TO 'app_developer
	CREATE USER 'dev'@'localhost
	GRANT 'app_developer' TO 'dev'@'localhost'
	SET DEFAULT ROLE 'app_developer' TO 'dev'@'localhost'
```

login as 'dev'@'localhost'

Before:
```
  mysql> use app_db
  ERROR 1044 (42000): Access denied for user 'dev'@'localhost' to database 'app_db'
```
After:
```
  mysql> use app_db;
  Database changed
```
### What is changed and how it works?

`USE DB` check privileges using the `DBIsVisible()` function, that function should take `role` into consideration

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @imtbkcat @c4pt0r

**Comment by tiancaiamao** [comment]
PTAL @jackysp @lysu

File: privilege/privileges/privileges.go [hunk 1]
```diff
@@ -160,12 +160,20 @@ func (p *UserPrivileges) ConnectionVerification(user, host string, authenticatio
...
 
 // DBIsVisible implements the Manager interface.
-func (p *UserPrivileges) DBIsVisible(db string) bool {
+func (p *UserPrivileges) DBIsVisible(activeRoles []*auth.RoleIdentity, db string) bool {
 	if SkipWithGrant {
 		return true
 	}
 	mysqlPriv := p.Handle.Get()
-	return mysqlPriv.DBIsVisible(p.user, p.host, db)
+	if mysqlPriv.DBIsVisible(p.user, p.host, db) {
```
Comment (imtbkcat) [inline privilege/privileges/privileges.go]: If `r_1` has privilege, `r_2` doesn't have, and has relationship like `r_1 -> r_2 -> user`, user should have privilege to visit db.

Comment (tiancaiamao) [reply privilege/privileges/privileges.go]: Do you mean user has role r_2
r_2 has role r_1 ?

I think `activeRoles` contains both r_1 and r_2 ? @imtbkcat

Comment (imtbkcat) [reply privilege/privileges/privileges.go]: `ctx.ActiveRoles` just contain r_1, you could use `MySQLPrivileges.FindAllRole` to get r_1 and r_2.

Comment (tiancaiamao) [reply privilege/privileges/privileges.go]: Done.
PTAL @imtbkcat

**Pull request merged**